### ISSUE TITLE
The .tar file loading process is quite unclear and takes up all the RAM

### ISSUE THREAD

**Issue by jboulanger91** [cajal/microns-nda-access#11]
Thanks for all the hard work putting all the data online. 
I'd like to load the v7.tar file using docker but the process is quite complicated for a docker newbie. 
I am on Win10. 

First when navigating to the .tar folder and executing the suggested: 
`docker-compose up -d database `
I get 
`ERROR: pull access denied for microns-phase3-nda-database, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`

Thne I ran: `docker run --network="host" --detach microns-phase3-nda-database:latest`
This opens a Vmmem process in my task manager which keeps loading the RAM with no direct effect.   

It is really unclear how the loading of the dataset will be linked to the notebooks for later data exploration. I am sorry if these questions are trivial but I'm sure they would help many users. 
Thanks a lot, 
jon

**Comment by Cpapa97** [comment]
Hi, glad to see someone trying to use the data!

I haven't tested this on Windows personally, but with Docker my advice should still work the same.
The first step should be to run
```console
docker load --input functional_data_database_container_image_v7.tar
```
in the folder where you've downloaded the .tar docker image.

Your error tells me that docker is looking for an image it can pull from dockerhub, but once you run the above command (which should take awhile) then it'll be able to find the image locally.

A bit more info on this part of the process is here: https://github.com/cajal/microns-nda-access#database

Let me know if this fixes your first problem and if you still have questions about the notebooks afterwards, I'm definitely interested to know if the existing instructions are sufficient so more users can use this data.

**Comment by jboulanger91** [comment]
I have run the command you suggested and I still get the memory overload problem. 
Could this be a Docker version or .tar file version issue? 
Maybe the v6 .tar will work but I don't know where to get it from. 
Thanks!

**Comment by Cpapa97** [comment]
The v6.tar isn't available anymore unfortunately, but I don't think that it would make a difference here since they were all built and tested the same way.

To go through the checklist again in case we missed anything,

You loaded the docker image from the v7.tar file. Does `microns-phase3-nda-database` show up when you run `docker image ls`?

Related, when you run `docker-compose up -d database` (from within the clone of the `microns-nda-access` repo, since at this point you're not using the .tar file anymore) do you still get the error `ERROR: pull access denied for microns-phase3-nda-database...`?
(for now I think it'll be easier to track down this issue if you're using docker-compose instead of docker run)

Another sanity check is this issue (https://github.com/cajal/microns-nda-access#known-issues) with not having enough memory pre-allocated to docker on Windows and Mac, though it's not related to the problem you're having as far as I know it might still be worth checking.

**Comment by jboulanger91** [comment]
Thanks for your reactivity. 

%%
I do `cd ./my dir` and then `dir` and I get:
08/15/2021  03:28 PM   109,365,078,528 functional_data_database_container_image_v7.tar
However when I do `docker image ls` I get : 
`REPOSITORY                        TAG                                        IMAGE ID       CREATED        SIZE
docker/dev-environments-default   stable-1                                   7c85b0303242   10 days ago    607MB
docker/desktop-git-helper         5a4fca126aadcd3f6cc3a011aa991de982ae7000   efe2d67c403b   2 weeks ago    44.2MB
redis                             latest                                     aa4d65e670d6   3 weeks ago    105MB
docker/getting-started            latest                                     083d7564d904   2 months ago   28MB`

So basically the full .tar doesn't show up anymore. I am re-downloading to see if the fil didn't get corrupted during the transfer. 

%%
When using the `docker-compose up -d database` form the micronsbinder env. I get : 
`Pulling database (microns-phase3-nda-database:latest)...
ERROR: error creating temporary lease: connection error: desc = "transport: Error while dialing dial unix /var/run/desktop-containerd/containerd.sock: connect: connection refused": unavailable`

%%
As suggested I have added a .wslconfig file increasing the mem to 12Gb. It is also unclear is this is enough as the .tar file is much larger then this ... 

Hope these answers were useful ...
Jon

**Comment by Cpapa97** [comment]
From what I can tell from my searching, and the best I can tell is that this specific error `ERROR: error creating temporary lease: connection error: desc = "transport: Error while dialing dial unix /var/run/desktop-containerd/containerd.sock: connect: connection refused": unavailable` is either caused by running out of disk space, inability to remove old containers, or something wrong with the docker daemon that should be fixable by restarting the daemon.

I'm not sure if this is related to the fact that `docker load` doesn't seem to be working, but I could believe that it is related if either the filesystem is out of disk space or the docker daemon is need of a restart, maybe even both.

So the first thing I would check is if somehow your disk ran out of space. The tar file being 110+ GBs means that in order to load the image into your docker image repository you need double that disk space, so you'd need at least 220+ GBs free. (Then if you do manage to get the image loaded I'd suggest deleting that .tar file to make space for the running container layer that lives on top of the completely-readonly layer of the original data.)

If the disk space is all in order then the next thing I would try is restarting your docker daemon. If this is within WSL (though I'm not quite familiar with how Docker for Windows works and if there's a difference with installing it through WSL or not, or how WSL 1.0 vs WSL 2.0 differences work), then I think you can use `service docker restart`, or the equivalent systemctl command, to restart it.

If none of this works and/or we can't figure out why you're unable to load the image from the .tar file, then I'll have to try to see if I can reproduce this (or find a way that works) and install Docker on my windows machine the same way you have it setup, and try to go through the same steps to see if I can pin down the cause of this issue. However, I can't guarantee when I'll be able to get around to doing that so I'll leave that as a last resort for now.

**Comment by jboulanger91** [comment]
Ok thanks for the feedback. The memory was quite full when I did` docker-compose`
After a restart I did: 
```
1. docker load --input functional_data_database_container_image_v7.tar
2. docker-compose up -d database
```
This took >6hrs total on a desktop Win10 PC. 

Then the instructions become (at least for me quite unclear) from the **Jupyter Notebook (DataJoint)**  part: 
 The folder architecture that needs to be used for the .tar; docker-compose.yml and the notebooks and ${EXTERNAL_NOTEBOOKS}: path is quite unclear. I cloned the cajal/microns_phase3_nda folder into my working directory containing the previous files and installed the packages.  

However when I launch `http://localhost:8888/tree` I can't navigate to the right folder containing the notebooks (the normal jupyter notebook path is used). I used a workaround and copied the notebooks in this path (which is on a different drive) but I couldn't get access to the to the functional data (After stetting the DataJoint permissions). It seems that the `%%capture` step can't proceed. I feel like the path management over multiple drives is not ideal ... 

Please let me know what I'm doing wrong. I feel like an example of the paths in the` .env` and in the docker_compose.yml would be very helpful. Thanks a lot again.

**Comment by Cpapa97** [comment]
Ah, I see where the instructions are misleading now.

This section in the README wasn't meant to point directly to the cajal/microns_phase3_nda repo, that was a mistake.
> You can clone this access repository and build it yourself with Docker and docker-compose. Clone the repository at https://github.com/cajal/microns_phase3_nda.

It's actually intended to use the same docker-compose (which has the notebook service) in this current repo (cajal/microns-nda-access) to access the notebooks and connect to the database with a default configuration. The path in the current `.env` actually points to a folder in this repo that'll be created by the Dockerfile and mounted automatically by the docker-compose service, so no alterations should be needed to get access to the notebooks now.

So if you've already clone this repo (cajal/microns-nda-access) and have started the `database` service with it, you should be able to just run `docker-compose up -d notebook` in the same directory and it'll build an image and then start the notebook container properly.

Thanks for bringing this issue up! I will definitely be fixing that confusing line.
